File: backend/app/schemas/cliente.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ClienteBase(BaseModel):
    nombre: str = Field(min_length=1, max_length=150)
    telefono: str | None = Field(default=None, max_length=30)
    email: str | None = Field(default=None, max_length=150)
    documento: str | None = Field(default=None, max_length=40)


class ClienteCreate(ClienteBase):
    pass


class ClienteUpdate(BaseModel):
    nombre: str | None = Field(default=None, min_length=1, max_length=150)
    telefono: str | None = Field(default=None, max_length=30)
    email: str | None = Field(default=None, max_length=150)
    documento: str | None = Field(default=None, max_length=40)


class ClienteOut(ClienteBase):
    id: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/cliente.ts
export interface Cliente {
  id: number;
  nombre: string;
  telefono: string | null;
  email: string | null;
  documento: string | null;
  created_at: string;
}

export interface ClienteCreate {
  nombre: string;
  telefono?: string | null;
  email?: string | null;
  documento?: string | null;
}

export interface ClienteUpdate {
  nombre?: string;
  telefono?: string | null;
  email?: string | null;
  documento?: string | null;
}
